okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java

```diff
@@ -233,13 +234,28 @@ public Level getLevel() {
 
       if (!logBody || !HttpHeaders.hasBody(response)) {
         logger.log("<-- END HTTP");
-      } else if (bodyEncoded(response.headers())) {
+      } else if (bodyHasUnknownEncofing(response.headers())) {
         logger.log("<-- END HTTP (encoded body omitted)");
       } else {
         BufferedSource source = responseBody.source();
         source.request(Long.MAX_VALUE); // Buffer the entire body.
         Buffer buffer = source.buffer();
 
+        Long gzippedLength = null;
+        if ("gzip".equalsIgnoreCase(headers.get("Content-Encoding"))) {
+            gzippedLength = buffer.size();
+            GzipSource gzippedResponseBody = null;
+            try {
+                gzippedResponseBody = new GzipSource(buffer.clone());
+                buffer = new Buffer();
+                gzippedResponseBody.read(buffer, Long.MAX_VALUE);
```

Comment:
I think because of the source.request above, the read here is guaranteed to get everything, but it would be good to have confirmation of that from someone (or a pointer to a better way to go about it).

Reply:
It is, but I don't think it's safe to rely on. Maybe try `buffer.writeAll(gzippedResponseBody)` instead?

Reply:
Ah, that's what I'd been searching for on the source :)

Great - Done.